### PR TITLE
Skip the validation of YAML if task is being removed from catalog

### LINKED ISSUE #619: Do not run tests for removed tasks in PR
# Expected Behavior
If a task is removed, do not run its e2e tests (or not try to create the removed task).

# Actual Behavior
While removing a task from catalog, e2e test script tries to run its e2e tests and fails the e2e run upon not finding the tests (or the task definition itself).
Looks like the `all_tests` should be populated from the changes in the PR.
https://github.com/tektoncd/catalog/blob/8d53c6ecf02777795c363e25d8eed8b2c2af23e7/test/e2e-tests.sh#L60

# Steps to Reproduce the Problem

1. Remove a task from catalog and raise the PR and notice the e2e tests fails complain failing to find the (removed) task definition.
2.
3.

# Additional Info
See the failure https://tekton-releases.appspot.com/build/tekton-prow/pr-logs/pull/tektoncd_catalog/618/pull-tekton-catalog-integration-tests/1351434107615186946/ for https://github.com/tektoncd/catalog/pull/618

### PR DESCRIPTION
**Pull request by vinamra28**
# Changes

If a task is being removed from the catalog then also the CI runs the
check for validating the YAML and thus fails if the task manifest is not
found.

This PR checks for the task directory whether it exists or not and in the case
where task directory is not present then the function doesn't checks the
validation of the YAML. Example if we remove the task knctl-deploy then
the CI would skip the check for that pull request.

closes #619 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

**Comment by vinamra28** [comment]
Added one extra commit 7002fc7 in which I removed one task and modified one task for testing and it is working fine. Link to logs can be found [here](https://tekton-releases.appspot.com/build/tekton-prow/pr-logs/pull/tektoncd_catalog/621/pull-tekton-catalog-integration-tests/1352137493159350272/?log#log)

**Comment by vinamra28** [comment]
/cc @chmouel @vdemeester 
please take a look :slightly_smiling_face:

File: test/e2e-common.sh [hunk 1]
```diff
@@ -116,6 +116,11 @@ function test_yaml_can_install() {
         # remove /0.1/tests from end
         local testname=${runtestdir%%/*}
         runtest=${runtest//tests}
+
+        # in case a task is being removed then it's directory
+        # doesn't exists, so skip the test for YAML
+        [ ! -d "${runtest%%/*}/${testname}" ] && continue
```
Comment (vdemeester) [inline test/e2e-common.sh:122]: Should we log something ?

**Review by vdemeester — APPROVED** [review]

**Pull request merged**